hadoop-ozone/dist/src/shell/ozone/ozone-functions.sh

```diff
@@ -2512,6 +2525,23 @@ function ozone_parse_args
           ozone_exit_with_usage 1
         fi
       ;;
+      --validate)
+        shift
+        OZONE_VALIDATE_CLASSPATH=true
+        if [[ "${1}" == "classpath" || "${1}" == "--daemon" ]]; then
+          ((OZONE_PARSE_COUNTER=OZONE_PARSE_COUNTER+1))
+        elif [[ "${1}" == "continue" && "${2}" == "--daemon" ]]; then
```

Comment:
We should be able to provide options in any order.  Please process combination of options (`--validate` and `--daemon`) after the loop has completed.

Also, currently only the following (somewhat counterintuitive order) is supported:

```
$ ozone --validate classpath ozone-manager
Validating classpath file: hadoop-ozone/dist/target/ozone-1.3.0-SNAPSHOT/share/ozone/classpath/ozone-manager.classpath
...
```

The following commands should also be supported, but are currently failing:

```
$ ozone classpath ozone-manager --validate
unrecognized option
[main] INFO util.ExitUtil: Exiting with status 1: unrecognized option

$ ozone classpath --validate ozone-manager
ERROR: Classpath file descriptor hadoop-ozone/dist/target/ozone-1.3.0-SNAPSHOT/share/ozone/classpath/--validate.classpath is missing
```

Reply:
Thank you for reviewing @adoroszlai, noted these points.

Regarding the ordering of the options - just wanted to check if we should be going by the following usage convention or not:

```
Usage: ozone [OPTIONS] SUBCOMMAND [SUBCOMMAND OPTIONS]

  OPTIONS is none or any of:

--buildpaths                       attempt to add class files from build tree
--config dir                       Ozone config directory
--daemon (start|status|stop)       operate on a daemon
--debug                            turn on shell script debug mode
--help                             usage information
--hostnames list[,of,host,names]   hosts to use in worker mode
--hosts filename                   list of hosts to use in worker mode
--jvmargs arguments                append JVM options to any existing options defined in the OZONE_OPTS environment variable. Any defined in OZONE_CLIENT_OPTS will be append after these jvmargs
--loglevel level                   set the log4j level for this command
--validate (continue)              validates if all jars as indicated in the corresponding OZONE_RUN_ARTIFACT_NAME classpath file are present
--workers                          turn on worker mode

  SUBCOMMAND is one of:


    Admin Commands:

daemonlog     get/set the log level for each daemon
jmxget        get JMX exported values from NameNode or DataNode.

    Client Commands:

admin         Ozone admin tool
auditparser   runs audit parser tool
classpath     prints the class path needed for running ozone commands
debug         Ozone debug tool
dtutil        operations related to delegation tokens
envvars       display computed Hadoop environment variables
freon         runs an ozone data generator
fs            run a filesystem command on Ozone file system. Equivalent to 'hadoop fs'
genconf       generate minimally required ozone configs and output to ozone-site.xml in specified path
getconf       get ozone config values from configuration
insight       tool to get runtime operation information
s3            command line interface for s3 related operations
sh            command line interface for object store operations
tenant        command line interface for multi-tenant related operations
version       print the version

    Daemon Commands:

csi           run the standalone CSI daemon
datanode      run a HDDS datanode
om            Ozone Manager
recon         run the Recon service
s3g           run the S3 compatible REST gateway
scm           run the Storage Container Manager service

SUBCOMMAND may print help when invoked w/o parameters or with -h.
```

Usage: `ozone [OPTIONS] SUBCOMMAND [SUBCOMMAND OPTIONS]`

- `[OPTIONS]` = `--validate`
- `SUBCOMMAND` = `classpath`
- `[SUBCOMMAND OPTIONS]` `ozone-manager`

Command: `ozone --validate classpath ozone-manager`

Reply:
Oh, you are right, since `--validate` is listed as an option of the `ozone` command, not the `classpath` subcommand, we can keep it like this.  Thanks.

Reply:
However, the first part my comment still stands, `--daemon` and `--validate` are both options of `ozone`, could be in any order.

Reply:
Sure @adoroszlai, will be making the changes to process the combination of options (`--validate` and `--daemon`) post-completion of the loop in any order, thank you!